File: 003/Problem3.py
import time


def largest_prime_factor(n: int) -> int:
    i = 2
    while i * i <= n:
        if n % i:
            i += 1
        else:
            n //= i
    return n


def problem():
    print(largest_prime_factor(600851475143))


start = time.time()
for i in range(1000):
    problem()
end = time.time()
print("Function took " + str(end - start))

File: 003/Problem3.c
#include <stdio.h>
#include <time.h>

long long largestPrimeFactor(long long n) {
    long long i = 2;
    while (i * i <= n) {
        if (n % i) {
            i += 1;
        } else {
            n /= i;
        }
    }
    return n;
}

void problem() {
    long long ans = largestPrimeFactor(600851475143);
    printf("%lld\n", ans);
}

int main() {
    clock_t t;
    t = clock();
    for (int i = 0; i < 1000; i++) {
        problem();
    }
    t = clock() - t;
    double time_taken = ((double) t ) / CLOCKS_PER_SEC;
    printf("Function took %f \n", time_taken);
    return 0;
}
